Tests/StencilTests/EnvironmentSpec.swift

```diff
@@ -1,342 +1,392 @@
-          template = nil
-          includedTemplate = nil
-        }
-
-        func expectError(reason: String, token: String, includedToken: String,
-                         file: String = #file, line: Int = #line, function: String = #function) throws {
-          var expectedError = expectedSyntaxError(token: token, template: template, description: reason)
-          expectedError.stackTrace = [expectedSyntaxError(token: includedToken, template: includedTemplate, description: reason).token!]
-
-          let error = try expect(environment.render(template: template, context: ["target": "World"]),
-                                 file: file, line: line, function: function).toThrow() as TemplateSyntaxError
-          let reporter = SimpleErrorReporter()
-          try expect(reporter.renderError(error), file: file, line: line, function: function) == reporter.renderError(expectedError)
-        }
-
-        $0.it("reports syntax error in included template") {
-          template = Template(templateString: """
-          {% include "invalid-include.html" %}
-          """, environment: environment)
-          includedTemplate = try environment.loadTemplate(name: "invalid-include.html")
-
-          try expectError(reason: "Unknown filter 'unknown'. Found similar filters: 'uppercase'.",
-                          token: """
-                          include "invalid-include.html"
-                          """,
-                          includedToken: "target|unknown")
-        }
-
-        $0.it("reports runtime error in included template") {
-          let filterExtension = Extension()
-          filterExtension.registerFilter("unknown", filter: {  (_: Any?) in
-            throw TemplateSyntaxError("filter error")
-          })
-          environment.extensions += [filterExtension]
-
-          template = Template(templateString: """
-          {% include "invalid-include.html" %}
-          """, environment: environment)
-          includedTemplate = try environment.loadTemplate(name: "invalid-include.html")
-
-          try expectError(reason: "filter error",
-                          token: "include \"invalid-include.html\"",
-                          includedToken: "target|unknown")
-        }
+    it("reports rendering error in block") {
+      self.template = Template(
+        templateString: "{% block some %}{% customtag %}{% endblock %}",
+        environment: self.environment
+      )
+      try self.expectError(reason: "Custom Error", token: "customtag")
+    }
+  }
 
-      }
+  private func expectError(
+    reason: String,
+    token: String,
+    file: String = #file,
+    line: Int = #line,
+    function: String = #function
+  ) throws {
+    let expectedError = expectedSyntaxError(token: token, template: template, description: reason)
+
+    let error = try expect(
+      self.environment.render(template: self.template, context: ["names": ["Bob", "Alice"], "name": "Bob"]),
```

Comment:
extract context in a local variable and also move template and context parameters on separate lines?

Reply:
I'd prefer to keep this function in a single line though, same as expectation later.

Reply:
> extract context in a local variable
Could you clarify this bit?

Reply:
```
let context = ["names": ["Bob", "Alice"], "name": "Bob"])
self.environment.render(template: template, context: context, file: file, line: line, function: function)
```

Reply:
Ah ok.

> in a single line though
Problem is, and this in quite a few places, that we have absolutely gigantically long function calls/init/...

Thanks to swiftlint, we can enforce a hard limit on this, to improve readability. We can try keeping some things on one line, but here and there, we *will* have to split things up into multiple lines.

Reply:
yes, I agree, though having a function call that takes more than 5 lines probably is a sign that we should refactor it.

Reply:
True, but there's not much of a choice here, we're passing along parameters to another function 🤷‍♂️